x/ecocredit/migrations/v3/patch.go

```diff
@@ -0,0 +1,190 @@
+package v3
+
+import (
+	"context"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	basketapi "github.com/regen-network/regen-ledger/api/regen/ecocredit/basket/v1"
+	api "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// patchMigrate performs following migrations
+// - add reference id to existing projects
+// - update issuance date for credit batches
+// - update curator address for baskets
+func patchMigrate(ctx context.Context, sdkCtx sdk.Context, ss api.StateStore,
+	basketStore basketapi.StateStore, oldBatchDenomToNewDenomMap map[string]string) error {
+	if sdkCtx.ChainID() == "regen-1" {
+		return patchMainnet(ctx, ss, oldBatchDenomToNewDenomMap)
+	} else if sdkCtx.ChainID() == "regen-redwood-1" {
+		return patchRedwood(ctx, ss, basketStore, oldBatchDenomToNewDenomMap)
+	}
+
+	return nil
+}
+
+func patchMainnet(ctx context.Context, ss api.StateStore, oldBatchDenomToNewDenomMap map[string]string) error {
+	// project location -> reference-id
+	// CD-MN -> "" // TODO: add reference-id
+	// KE    -> ""
```

Comment:
@clevinson and I discussed reference id with @S4mmyb for the projects that be created but no reference id was decided upon. We can add later when we are closer to a final release if this is still undecided.

Reply:
See https://github.com/regen-network/regen-ledger/pull/1121#discussion_r888410762